single_kernel_mongo/state/charm_state.py

```diff
@@ -457,6 +458,21 @@ def set_scaling_down(self, rel_id: int, departing_unit_name: str) -> bool:
         self.unit_peer_data.update({rel_departed_key: json.dumps(scaling_down)})
         return scaling_down
 
+    def share_status_with_config_server(self, status: StatusBase):
+        """Shares this shard's status with the config server."""
```

Comment:
```suggestion
        """
        Shares this shard's status with the config server.
        
        This is primarily useful for when the cluster upgrades, since the config-server will need to ensure all units are healthy.
        """
```